servers/src/main/java/tachyon/worker/block/BlockDataManager.java

```diff
@@ -0,0 +1,455 @@
+  private final WorkerSource mWorkerSource;
+  /** Metrics reporter that listens on block events and increases metrics counters*/
+  private final BlockMetricsReporter mMetricsReporter;
+
+  // TODO: See if this can be removed from the class
+  /** MasterClient, only used to inform the master of a new block in commitBlock */
+  private MasterClient mMasterClient;
+  /** UnderFileSystem Client */
+  private UnderFileSystem mUfs;
+  /** User metadata, used to keep track of user heartbeats */
+  private Users mUsers;
+  /** Id of this worker */
+  long mWorkerId;
+
+  /**
+   * Creates a BlockDataManager based on the configuration values.
+   *
+   * @param tachyonConf the configuration values to use
+   * @param workerSource object for collecting the worker metrics
+   * @throws IOException if the tiered store fails to initialize
+   */
+  public BlockDataManager(TachyonConf tachyonConf, WorkerSource workerSource) throws IOException {
+    mHeartbeatReporter = new BlockHeartbeatReporter();
+    mBlockStore = new TieredBlockStore(tachyonConf);
+    mTachyonConf = tachyonConf;
+    mWorkerSource = workerSource;
+    mMetricsReporter = new BlockMetricsReporter(mWorkerSource);
+
+    mMasterClientExecutorService =
+        Executors.newFixedThreadPool(1,
+            ThreadFactoryUtils.build("worker-client-heartbeat-%d", true));
+    mMasterClient =
+        new MasterClient(getMasterAddress(), mMasterClientExecutorService, mTachyonConf);
+
+    // Create Under FileSystem Client
+    String tachyonHome = mTachyonConf.get(Constants.TACHYON_HOME, Constants.DEFAULT_HOME);
+    String ufsAddress =
+        mTachyonConf.get(Constants.UNDERFS_ADDRESS, tachyonHome + "/underFSStorage");
+    mUfs = UnderFileSystem.get(ufsAddress, mTachyonConf);
+    // Connect to UFS to handle UFS security
+    InetSocketAddress workerAddress = getWorkerAddress();
+    try {
+      mUfs.connectFromWorker(mTachyonConf, NetworkUtils.getFqdnHost(workerAddress));
+    } catch (IOException e) {
+      LOG.error("Worker @ " + workerAddress + " failed to connect to the under file system", e);
+      throw Throwables.propagate(e);
+    }
+
+    // Register the heartbeat reporter so it can record block store changes
+    mBlockStore.registerBlockStoreEventListener(mHeartbeatReporter);
+    mBlockStore.registerBlockStoreEventListener(mMetricsReporter);
+  }
+
+  /**
+   * Aborts the temporary block created by the user.
+   *
+   * @param userId The id of the client
+   * @param blockId The id of the block to be aborted
+   * @throws IOException if the block does not exist
+   */
+  public void abortBlock(long userId, long blockId) throws IOException {
+    mBlockStore.abortBlock(userId, blockId);
+  }
+
+  /**
+   * Access the block for a given user. This should be called to update the evictor when necessary.
+   *
+   * @param userId The id of the client
+   * @param blockId The id of the block to access
+   */
+  public void accessBlock(long userId, long blockId) throws IOException {
```

Comment:
Oh its actually not thrown since we are using the tieredblockstore, but I will add a comment.